cfut/__init__.py

```diff
@@ -52,22 +57,27 @@ def wait(self, filename, value):
             self.waiting[filename] = value
 
     def run(self):
-        while True:
-            with self.lock:
-                if self.shutdown:
-                    return
-
-                # Poll for each file.
-                for filename in list(self.waiting):
-                    if os.path.exists(filename):
-                        self.callback(self.waiting[filename])
-                        del self.waiting[filename]
+        for i in count():
+            if self.shutdown:
+                return
 
+            self.check(i)
             time.sleep(self.interval)
 
+    def check(self, i):
```

Comment:
Perhaps a docstring would be helpful here—I was initially confused as to why `i` was passed as an argument, but I see it now after reading the override in `SlurmWaitThread`.